enterprise/backend/src/metabase_enterprise/sso/integrations/sso_utils.clj

```diff
@@ -25,15 +26,24 @@
    ;; TODO - we should avoid hardcoding this to make it easier to add new integrations. Maybe look at something like
    ;; the keys of `(methods sso/sso-get)`
    [:sso_source       [:enum :saml :jwt]]
+   [:groups {:optional true} [:maybe seqable?]]
    [:login_attributes [:maybe :map]]])
 
 (mu/defn create-new-sso-user!
   "This function is basically the same thing as the `create-new-google-auth-user` from `metabase.models.user`. We need
   to refactor the `core_user` table structure and the function used to populate it so that the enterprise product can
   reuse it"
   [user :- UserAttributes]
-  (u/prog1 (first (t2/insert-returning-instances! User (merge user {:password (str (random-uuid))})))
+  (u/prog1 (first (t2/insert-returning-instances! User (merge (dissoc user :groups) {:password (str (random-uuid))})))
     (log/info (trs "New SSO user created: {0} ({1})" (:common_name <>) (:email <>)))
+    ;; publish user-invited event for audit logging
+    (let [details (-> (into {} <>)
+                      (dissoc :last_login :is_qbnewb :is_superuser :date_joined :common_name :invitor)
+                      (merge {:invitor       nil
+                              :invite_method "sso"
+                              :groups (:groups user)}))]
+      (events/publish-event! :event/user-invited {:user-id api/*current-user-id*
```

Comment:
Not sure if `*current-user-id*` will be bound here. Regardless, the audit infra will automatically read `*current-user-id*` and write it in the audit_log table if it exists.

I think for the `user-invited` event we could have
* `user_id` column containing the ID of the user who initiated the invite (or `nil` if it's a user being created via SSO)
* `id` in the `details` map containing the ID of the user who is being created